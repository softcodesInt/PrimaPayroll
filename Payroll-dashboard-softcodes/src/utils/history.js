/* -------------------------------------------------------------------------- */
/*                             External Dependeny                             */
/* -------------------------------------------------------------------------- */
import { createBrowserHistory } from 'history';

export default createBrowserHistory();
