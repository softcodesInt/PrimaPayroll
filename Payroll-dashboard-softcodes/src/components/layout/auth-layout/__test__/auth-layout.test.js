/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import AuthLayout from '..';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import renderWithRouter from 'utils/render-with-router';
import { Router } from 'react-router-dom';

/* -------------------------- Internal Dependencies ------------------------- */
import history from '../../../../redux/history';
import isAuthenticated from '../../../../utils/is-authenticated';

afterEach(cleanup);

describe('AuthLayout Component', () => {
	it('AuthLayout Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Router history={history}>
				<AuthLayout />
			</Router>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('AuthLayout Renders Correctly With Attributes', () => {
		const { getByTestId } = render(
			<AuthLayout>
				<h1>Auth Layout</h1>
			</AuthLayout>
		);
		expect(getByTestId('auth-layout')).toHaveTextContent('Auth Layout');
	});

	it('AuthLayout matches snapshot', () => {
		const tree = renderer.create(<AuthLayout>AuthLayout</AuthLayout>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('redirects when authToken invalid', async () => {
		const { history: historys } = renderWithRouter(<AuthLayout />);
		if (!isAuthenticated()) {
			expect(historys.location.pathname).toEqual('/');
		}
	});
});
