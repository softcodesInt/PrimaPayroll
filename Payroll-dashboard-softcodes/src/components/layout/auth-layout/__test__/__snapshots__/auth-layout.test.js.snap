// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout Component AuthLayout matches snapshot 1`] = `
<div
  className="sc-bdnylx TyOEk"
>
  <div
    className="grid_layout d-block d-md-grid"
  >
    <div
      className="grid-1 d-flex"
    >
      <div
        className="child__item container"
        data-testid="auth-layout"
      >
        <svg
          className="mb-4"
        >
          logos.png
        </svg>
        AuthLayout
      </div>
    </div>
    <div
      className="grid-2 d-none d-md-block"
    >
      <img
        alt="layout banner"
        src="Body.png"
      />
    </div>
  </div>
</div>
`;
